### ISSUE TITLE
Working only after opening the GUI

### ISSUE THREAD

**Issue by iNewLegend** [withfig/fig#2367]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [ ] I have run `fig restart` and replicated the issue again

### Operating system

macOS 13.2.1 (22D68)

### Expected behaviour

To fig work out of box

### Actual behaviour

I have to open the GUI, so terminal autocomplete begins to work

### Steps to reproduce

The issue started after upgrade the os to
<img width="470" alt="image" src="https://user-images.githubusercontent.com/10234691/221124621-80d0069d-2bb1-4ed0-92fc-3897886ae692.png">

### Environment

```yaml
fig-details:
  - 2.14.2
hardware-info:
  - model: 
  - model-id: 
  - chip-id: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  - cores: 6
  - mem: 16.00 GB
os-info:
  - macOS 13.2.1 (22D68)
environment:
  - shell: /bin/zsh
  - terminal: iterm
  - cwd: /Users/inewlegend/Desktop/zenflux/demo-shop-catalog
  - exe-path: /usr/local/bin/fig
  - install-method: brew
  - env-vars:
    - FIGTERM_SESSION_ID: 48eaf3da-38d3-47b2-a21f-5a855f8837df
    - FIG_SET_PARENT_CHECK: 1
    - FIG_TERM: 2.14.2
    - PATH: /usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/inewlegend/.fig/bin:/Users/inewlegend/.local/bin
    - SHELL: /bin/zsh
    - TERM: xterm-256color
    - __CFBundleIdentifier: com.googlecode.iterm2
    - FIG_PID: 14717
    - FIG_SET_PARENT: 48eaf3da-38d3-47b2-a21f-5a855f8837df
```
